api/src/pcapi/core/offers/validation.py

```diff
@@ -111,32 +112,34 @@ def check_stock_price(price: decimal.Decimal, offer: models.Offer) -> None:
             "Le prix d’une offre ne peut excéder 300 euros.",
         )
         raise errors
-    if finance_repository.has_active_or_future_custom_reimbursement_rule(offer):
-        # We obviously look for active rules, but also future ones: if
-        # a reimbursement rule has been negotiated that will enter in
-        # effect tomorrow, we don't want to let the offerer change its
-        # price today.
-        error = (
-            "Vous ne pouvez pas modifier le prix ou créer un stock pour cette offre, "
-            "car elle bénéficie d'un montant de remboursement spécifique."
-        )
-        current_prices = {stock.price for stock in offer.stocks if not stock.isSoftDeleted}
-        if len(current_prices) > 1:
-            # This is not supposed to happen, we should be notified.
-            logger.error(
-                "An offer with a custom reimbursement rule has multiple prices",
-                extra={
-                    "offer": offer.id,
-                    "prices": current_prices,
-                },
+    if not getattr(flask.request, "_cached_checked_custom_reimbursment_rules", False):
```

Comment:
Après réflexion, on devrait peut-être être plus prudent et indiquer, dans le nom du cache, l'id de l'offre. Ainsi, si jamais on créé des stocks pour **plusieurs** offres dans la même requête, on aura un cache par offre.

```suggestion
    cache_attribute = f"_cached_checked_custom_reimbursement_rules_{offer.id}
    if not getattr(flask.request, cahe_attribute, False):
```

.. et réutilisation de `cache_attribute` à la fin du bloc quand on met sa valeur à `True`.